### PR TITLE
Implemented states for main view

### PR DESCRIPTION
**Pull request by yukim17**

### CODE REVIEW

File: RayD/RayD/UI/MainView/MainViewModel.swift [hunk 1]
```diff
@@ -108,18 +141,35 @@ final class MainViewModel: ObservableObject {
         }
     }
 
-    func saveUncompletedSession() {
-        // save session only when session is running, because when user stops it manually it will be saved in Core Data
-        guard state == .run, vitDTracker.todayVitD > 0 else { return }
+    private func startSession() {
```
[the remark this answers is not in the document]
Comment (slavny11) [reply RayD/RayD/UI/MainView/MainViewModel.swift:144]: But maybe in the next edition

**Review by GabriellAnnunziata — APPROVED** [review]

File: RayD/RayD/UI/MainView/MainViewModel.swift [hunk 2]
```diff
@@ -148,7 +200,7 @@ final class MainViewModel: ObservableObject {
             self.vitDTracker.updateTodayVitD()
 
             if self.secondsToCompletion < 0 {
```
Comment (slavny11) [inline RayD/RayD/UI/MainView/MainViewModel.swift:202]: Shouldn't it be <= 0?

File: RayD/RayD/UI/MainView/States/StateMachine.swift
Comment (slavny11) [inline RayD/RayD/UI/MainView/States/StateMachine.swift:1]: blia, her nogu slomit🔥

**Review by slavny11 — APPROVED** [review]
ohuen4ik

File: RayD/RayD/UI/MainView/MainViewModel.swift (hunk 2 above)
Comment (yukim17) [reply RayD/RayD/UI/MainView/MainViewModel.swift:202]: I don't know, I just leave this part as it was before

**Pull request closed without merging**